nullaway/src/main/java/com/uber/nullaway/handlers/AssertionHandler.java

```diff
@@ -0,0 +1,128 @@
+import com.uber.nullaway.dataflow.AccessPathNullnessPropagation;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+
+/** This Handler deals with assertions which ensure that their arguments cannot be null. */
+public class AssertionHandler extends BaseNoOpHandler {
+  public AssertionHandler() {
+    super();
+  }
+
+  @Override
+  public NullnessHint onDataflowVisitMethodInvocation(
+      MethodInvocationNode node,
+      Types types,
+      Context context,
+      AccessPathNullnessPropagation.SubNodeValues inputs,
+      AccessPathNullnessPropagation.Updates thenUpdates,
+      AccessPathNullnessPropagation.Updates elseUpdates,
+      AccessPathNullnessPropagation.Updates bothUpdates) {
+    Symbol.MethodSymbol callee = ASTHelpers.getSymbol(node.getTree());
+    if (callee == null) {
+      return NullnessHint.UNKNOWN;
+    }
+
+    if (!areMethodNamesInitialized()) {
+      initializeMethodNames(callee);
+    }
+
+    /**
+     * Look for statements of the form: assertThat(A).isNotNull()
+     *
+     * <p>A will not be NULL after this statement.
+     */
+    if (isMethodIsNotNull(callee)) {
+      Node receiver = node.getTarget().getReceiver();
+      if (receiver instanceof MethodInvocationNode) {
+        MethodInvocationNode receiver_method = (MethodInvocationNode) receiver;
+        Symbol.MethodSymbol receiver_symbol = ASTHelpers.getSymbol(receiver_method.getTree());
+        if (isMethodAssertThat(receiver_symbol)) {
+          Node arg = receiver_method.getArgument(0);
+          AccessPath ap = AccessPath.getAccessPathForNodeNoMapGet(arg);
+          if (ap != null) {
+            bothUpdates.set(ap, NONNULL);
+          }
+        }
+      }
+    }
+    return NullnessHint.UNKNOWN;
+  }
+
+  private boolean isMethodIsNotNull(Symbol.MethodSymbol methodSymbol) {
+    return matchesMethod(methodSymbol, isNotNull, isNotNullOwner);
+  }
+
+  private boolean isMethodAssertThat(Symbol.MethodSymbol methodSymbol) {
+    return matchesMethod(methodSymbol, assertThat, assertThatOwner);
+  }
+
+  private boolean matchesMethod(
+      Symbol.MethodSymbol methodSymbol, Name toMatchMethodName, Name toMatchOwnerName) {
+    return methodSymbol.name.equals(toMatchMethodName)
+        && methodSymbol.owner.getQualifiedName().equals(toMatchOwnerName);
+  }
+
+  private boolean areMethodNamesInitialized() {
+    return isNotNull != null;
+  }
+
+  private synchronized void initializeMethodNames(Symbol.MethodSymbol methodSymbol) {
+    isNotNull = methodSymbol.name.table.fromString(IS_NOT_NULL_METHOD);
+    isNotNullOwner = methodSymbol.name.table.fromString(IS_NOT_NULL_OWNER);
+    assertThat = methodSymbol.name.table.fromString(ASSERT_THAT_METHOD);
+    assertThatOwner = methodSymbol.name.table.fromString(ASSERT_THAT_OWNER);
+  }
+
+  /**
+   * Strings corresponding to the names of the methods (and their owners) used to identify
+   * assertions in this handler.
+   */
```

Comment:
These two should probably also be normal comments and not javadoc unless you want to add one per field :) (as it is, javadoc will associate this comment with `IS_NOT_NULL_METHOD` only and consider the other fields to not have docs.

Reply:
Agreed

Reply:
Updated the comments. I didn't quite realize these comments become javadoc.